### PR TITLE
Update Terraform Docs Landing Page

### PR DESCRIPTION
**Pull request by laurapacilio**
## 🔗 Relevant links
- [Preview link](https://dev-portal-qvwajizdy-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/0/1202785022290115/f) 🎟️

## 🗒️ What

This PR updates the Terraform documentation landing page for the devdot launch. It attempts to replicate the content we have on this page: https://www.terraform.io/docs

## 🤷 Why

The docs need a landing page. :) And our decision with Judith is that we want to replicate the original landing page as much as possible, since we like it and so that we don't reinvent the wheel.

### Current docs landing page
Screen Shot 2022-09-02 at 3 44 23 PM

### Docs landing page in devdot - PR updates

Screen Shot 2022-09-06 at 5 31 29 PM

### CODE REVIEW

File: src/content/terraform/docs-landing.json [hunk 1]
```diff
@@ -1,110 +1,82 @@
 {
-	"pageSubtitle": "Terraform is like Vault, but different. Vault is an identity-based secret and encryption management system. This documentation covers the main concepts of Vault, what problems it can solve, and contains a quick start for using Vault.",
+	"pageSubtitle": "Terraform is an infrastructure as code tool that lets you build, change, and version infrastructure safely and efficiently.",
```
Comment (laurapacilio) [inline src/content/terraform/docs-landing.json:2]:
```suggestion
	"pageSubtitle": "Terraform is an infrastructure as code tool that lets you build, change, and version infrastructure safely and efficiently. This includes both low-level components like compute instances, storage, and networking, as well as high-level components like DNS entries and SaaS features.",
```

File: src/content/terraform/docs-landing.json [hunk 2]
```diff
@@ -1,110 +1,82 @@
...
 			]
 		},
 		{
-			"type": "section-heading",
-			"title": "Developers"
-		},
-		{
-			"type": "icon-card-grid",
+			"type": "card-grid",
+			"title": "Collaborate",
```
Comment (judithpatudith) [inline src/content/terraform/docs-landing.json:64]: I think this should be "Develop and Share"?

File: src/content/terraform/docs-landing.json [hunk 3]
```diff
@@ -1,110 +1,82 @@
...
+					"title": "CDK for Terraform",
+					"url": "/terraform/cdktf"
 				}
 			]
 		},
 		{
 			"type": "card-grid",
-			"title": "Key Management",
-			"description": "Use a standardized workflow for distribution and lifecycle management of cryptographic keys in various KMS providers.",
+			"title": "Develop and Share",
```
Comment (judithpatudith) [inline src/content/terraform/docs-landing.json:48]: I think this one should be "Collaborate"? for some reason I'm not sure the preview is up to date with the file I'm seeing here 🤔

**Review by judithpatudith — APPROVED** [review]
Approving to unblock but I left a couple comments. Thanks!

File: src/content/terraform/docs-landing.json (hunk 3 above)
Comment (laurapacilio) [reply src/content/terraform/docs-landing.json:48]: Yes, I've been having some trouble with this PR. I'm so sorry - let me get this all fixed up and I'll ping you with a fresh preview so you can more easily check it

Comment (laurapacilio) [inline src/content/terraform/docs-landing.json:48]:
```suggestion
			"title": "Collaborate",
```

File: src/content/terraform/docs-landing.json (hunk 2 above)
Comment (laurapacilio) [reply src/content/terraform/docs-landing.json:64]: yeah .....i don't know what happened here - thank you for catching this!

File: src/content/terraform/docs-landing.json [hunk 4]
```diff
@@ -1,110 +1,82 @@
...
 			]
 		},
 		{
-			"type": "section-heading",
-			"title": "Developers"
-		},
-		{
-			"type": "icon-card-grid",
+			"type": "card-grid",
+			"title": "Collaborate",
```
Comment (laurapacilio) [inline src/content/terraform/docs-landing.json:64]:
```suggestion
			"title": "Develop and Share",
```

File: src/content/terraform/docs-landing.json [hunk 5]
```diff
@@ -1,110 +1,82 @@
...
-					"url": "/vault/docs/secrets/kv"
+					"description": "Terraform creates and manages resources on cloud platforms and other services through their application programming interfaces (APIs).",
+					"title": "What is Terraform?",
+					"url": "/terraform/intro"
 				},
 				{
-					"description": "Generate database credentials dynamically based on configured roles.",
-					"title": "Database Credentials",
-					"url": "/vault/docs/secrets/databases"
+					"description": "Learn about popular Terraform use cases and related documentation you can use to create Terraform configurations and workflows.",
```
Comment (laurapacilio) [inline src/content/terraform/docs-landing.json:14]:
```suggestion
					"description": "Popular use cases and related documentation you can use to create Terraform configurations and workflows.",
```

File: src/content/terraform/docs-landing.json [hunk 6]
```diff
@@ -1,110 +1,82 @@
...
 			"type": "card-grid",
-			"title": "Secrets Management",
-			"description": "Centrally store, access, and deploy secrets across applications, systems, and infrastructure.",
+			"title": "Introduction to Terraform",
 			"cards": [
 				{
-					"description": "A generic Key-Value store used to store arbitrary secrets within the configured physical storage for Vault.",
-					"title": "KV Secrets Engine",
-					"url": "/vault/docs/secrets/kv"
+					"description": "Terraform creates and manages resources on cloud platforms and other services through their application programming interfaces (APIs).",
```
Comment (laurapacilio) [inline src/content/terraform/docs-landing.json:9]:
```suggestion
					"description": "How Terraform solves infrastructure challenges.",
```

File: src/content/terraform/docs-landing.json [hunk 7]
```diff
@@ -1,110 +1,82 @@
...
-		},
-		{
-			"type": "section-heading",
-			"title": "Use Cases"
-		},
 		{
 			"type": "card-grid",
-			"title": "Secrets Management",
-			"description": "Centrally store, access, and deploy secrets across applications, systems, and infrastructure.",
+			"title": "Introduction to Terraform",
```
Comment (laurapacilio) [inline src/content/terraform/docs-landing.json:6]:
```suggestion
			"title": "Introduction",
```

**Pull request merged**